moto/dynamodb2/models/__init__.py

```diff
@@ -138,6 +138,14 @@ def update(
                     value = self._get_default(value)
                     # If operation == list_append, get the original value and append it
                     value = self._get_appended_list(value, expression_attribute_values)
+                    # If there is a +-operator, we need to compute the new value
```

Comment:
Will also be covered by structured parsing but would be nice if we merge this such that we make sure we cover these cases as well.